### PR TITLE
Hw4 task 1 - read-file implementation + tests

### PR DESCRIPTION
**Pull request by IuliiaLanchuk**
I have 3 the same fixtures but with different parameters for positive, negative and exception tests.  I tried to do one common fixture, but it works incorrectly. Is it possible to do one and should I try to fix it?

### CODE REVIEW

File: homework4/task1/project1_read_file/read_magic_number.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+* https://en.wikipedia.org/wiki/Interval_(mathematics)#Terminology
+** https://docs.python.org/3/library/os.path.html
+*** https://docs.python.org/3/tutorial/errors.html#handling-exceptions
+**** https://docs.python.org/3/tutorial/errors.html#raising-exceptions
+"""
+
+
+def read_magic_number(path: str) -> bool:
+    """Return True if in first line there is a number in an interval [1, 3)* and False otherwise."""
```
Comment (Cjkjvfnby) [inline homework4/task1/project1_read_file/read_magic_number.py]: `*` is a reference to a footnote. Do you have a footnote here?

File: homework4/task1/tests/test_read_magic_number.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    file_content = request.param
+    file_path = directory / "test_data.txt"
+    file_path.write_text(file_content)
+    created_paths.add(file_path)
+    yield file_path
+    file_path.unlink()
+    directory.rmdir()
+
+
+def test_read_magic_number_positive(positive_test_data_file_creation):
```
Comment (Cjkjvfnby) [inline homework4/task1/tests/test_read_magic_number.py]: `positive_test_data_file_creation`  it is not a creation it is a path.

File: homework4/task1/tests/test_read_magic_number.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+def test_read_magic_number_exception(exception_test_data_file_creation):
+    file_path = exception_test_data_file_creation
+    assert file_path.exists() is True
+
+    with pytest.raises(ValueError, match="In first line a number is required."):
+        assert read_magic_number(str(file_path)) is None
+
+
+def test_datafiles_already_removed():
```
Comment (Cjkjvfnby) [inline homework4/task1/tests/test_read_magic_number.py]: You are not supposed to test it.  It is the behavior of the test itself and we don't test tests.

File: homework4/task1/tests/test_read_magic_number.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+def exception_test_data_file_creation(tmp_path, request):
+    directory = tmp_path / "dir_for_tests"
+    directory.mkdir()
+    file_content = request.param
+    file_path = directory / "test_data.txt"
+    file_path.write_text(file_content)
+    created_paths.add(file_path)
+    yield file_path
+    file_path.unlink()
+    directory.rmdir()
```
Comment (Cjkjvfnby) [inline homework4/task1/tests/test_read_magic_number.py]: Do we really need to cleanup `tmp_path` by our self?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
It is ok for me to have 3 fixtures.  

In general, it is solved by "factory".  Your fixture returns a function that accepts content and returns a path.

I left some minor comments.

File: homework4/task1/tests/test_read_magic_number.py (hunk 4 above)
Comment (IuliiaLanchuk) [reply homework4/task1/tests/test_read_magic_number.py]: As It is temporary, so, really we don`t need to cleanup it by ourselves.  I found thet those temporary files will be deleted automatically by application once their process is got completed.
By default /tmp directory is cleaned up only at the time of system startup or reboot.
By default applications are automatically delete their contents from this directory once their process is got completed.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**